src/snowflake/cli/plugins/snowpark/package/commands.py

```diff
@@ -131,46 +133,127 @@ def package_upload(
     "-y",
     hidden=True,
     help="Installs packages that are not available on the Snowflake Anaconda channel.",
+    callback=deprecated_flag_callback(
+        "Using --yes is deprecated. Create command always checks for package in PyPi."
+    ),
+)
+
+deprecated_allow_native_libraries_option = typer.Option(
+    PypiOption.NO.value,
+    "--allow-native-libraries",
+    help="Allows native libraries, when using packages installed through PIP",
+    hidden=True,
+    callback=deprecated_flag_callback_enum(
+        "--allow-native-libraries flag is deprecated. Use --allow-shared-libraries flag instead."
+    ),
+)
+
+ignore_anaconda_option = typer.Option(
+    False,
+    "--ignore-anaconda",
+    help="Does not lookup packages on Snowflake Anaconda channel.",
+)
+
+index_option = typer.Option(
+    None,
+    "--index-url",
+    help="Base URL of the Python Package Index to use for package lookup. This should point to "
+    " a repository compliant with PEP 503 (the simple repository API) or a local directory laid"
+    " out in the same format.",
+    show_default=False,
+)
+
+skip_version_check_option = typer.Option(
+    False,
+    "--skip-version-check",
+    help="Skip comparing versions of dependencies between requirements and Anaconda.",
+)
+
+allow_shared_libraries_option = typer.Option(
+    False,
+    "--allow-shared-libraries",
+    help="Allows shared (.so) libraries, when using packages installed through PIP.",
 )
 
 
 @app.command("create", requires_connection=True)
-@cleanup_after_install
+@cleanup_packages_dir
 def package_create(
     name: str = typer.Argument(
         ...,
         help="Name of the package to create.",
     ),
-    install_packages: bool = install_option,
+    ignore_anaconda: bool = ignore_anaconda_option,
+    index_url: Optional[str] = index_option,
+    skip_version_check: bool = skip_version_check_option,
+    allow_shared_libraries: bool = allow_shared_libraries_option,
+    deprecated_allow_native_libraries: PypiOption = deprecated_allow_native_libraries_option,
     _deprecated_install_option: bool = deprecated_install_option,
-    allow_native_libraries: PypiOption = PackageNativeLibrariesOption,
+    _deprecated_install_packages: bool = deprecated_pypi_download_option,
     **options,
 ) -> CommandResult:
     """
     Creates a Python package as a zip file that can be uploaded to a stage and imported for a Snowpark Python app.
     """
-    if _deprecated_install_option:
-        install_packages = _deprecated_install_option
+    allow_shared_libraries_pypi_option = {
+        True: PypiOption.YES,
```

Comment:
```suggestion
        True: YesNoAsk.YES,
```
Consider renaming this option, even if we are deprecating it